### PR TITLE
Typecheck engagement_edge

### PR DESCRIPTION
**Pull request by wimax-grapl**
My next task ( https://github.com/grapl-security/issue-tracker/issues/115 ) has me editing Engagement Edge.

I like static types. 

I also discovered many unused functions in here.

### CODE REVIEW

File: dobi.yaml [hunk 1]
```diff
@@ -555,6 +555,20 @@ job=typecheck-model-plugin-deployer:
       pip install mypy &&
       mypy .
       "
+
+job=typecheck-engagement-edge:
+  use: engagement-edge-build
+  # since there's no setup.py, I have an explicit `pip install mypy` instead of `.[typecheck]`
```
Comment (wimax-grapl) [inline dobi.yaml:561]: this comment is wrong and should be removed

File: src/python/engagement_edge/src/engagement_edge.py [hunk 2]
```diff
@@ -84,292 +101,15 @@
     )
```
Comment (wimax-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:103]: there was a TON of unused code in this service! any idea what's up with that? should i keep it around for a rainy day or something?

File: src/python/engagement_edge/src/engagement_edge.py [hunk 3]
```diff
@@ -418,17 +160,18 @@ def get_salt_and_pw(table, username):
         LOGGER.debug(f"Did not get salt for user: {username}")
         return None, None
 
-    salt = response["Item"]["salt"].value
-    password = response["Item"]["password"]
+    # Not quite sure what type this is supposed to be.
+    salt = response["Item"]["salt"].value  # type: ignore
```
Comment (wimax-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:164]: basically, mypy doesn't think there's anything that could be in here that responds to `.value`

File: src/python/engagement_edge/src/engagement_edge.py [hunk 4]
```diff
@@ -444,23 +187,7 @@ def user_auth_table():
     return DYNAMO.Table(os.environ["USER_AUTH_TABLE"])
 
 
-def create_user(username, cleartext):
```
Comment (wimax-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:447]: removing this, in particular, freaked me out - but searching by the pepper, it looks like the only palces we create users now are
- grapl provision (the service)
- grapl provision (the ipynb)

File: src/python/engagement_edge/src/engagement_edge.py [hunk 5]
```diff
@@ -512,21 +239,24 @@ def lambda_login(event):
...
-def requires_auth(path):
+def requires_auth(path: str) -> Callable[[RouteFn], RouteFn]:
     if not IS_LOCAL:
         path = "/{proxy+}" + path
 
-    def route_wrapper(route_fn):
+    def route_wrapper(route_fn: RouteFn) -> RouteFn:
         @app.route(path, methods=["OPTIONS", "POST"])
-        def inner_route():
+        def inner_route() -> Response:
```
Comment (wimax-grapl) [inline src/python/engagement_edge/src/engagement_edge.py:259]: this seems like it should prob forward args/kwargs, same with `no_auth`?

**Review by inickles-grapl — APPROVED** [review]

**Pull request merged**